app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -60,6 +62,11 @@ def deployment_spec(opts)
     deploy_options
   end
 
+  def workflow_helper
+    values = {:src_vm_id => [id], :src_vm_class_name => "OrchestrationTemplate"}
+    @workflow_helper ||= ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.new(values, User.current_user, :skip_dialog_load => true)
```

Comment:
@lfu won't this hit this `VmOrTemplate.find_by()` with the id of an orchestration template? https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_provision_virt_workflow.rb#L17

Reply:
It is possible `initial_pass` might not be true but that still feels very shaky

Reply:
@lfu This seems a bit fragile, based on the above comments. It would be much safer if we had a new class dedicated to provisioning from `OrchestrationTemplates` that would inherit from this one.

Something like  - `class MiqProvisionOrchWorkflow < MiqProvisionVirtWorkflow`

What do you think?